File: MyMoneyApi/MyMoneyApi/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace MyMoneyApi.Models;

public class Transaction
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    
    [Required]
    public string Title { get; set; } = string.Empty;
    
    [Required]
    public decimal Amount { get; set; }
    
    [Required]
    public string Category { get; set; } = string.Empty;
    
    public DateTime Date { get; set; } = DateTime.UtcNow;
    
    [Required]
    public TransactionType Type { get; set; }
    
    public string? Description { get; set; }
    
    public string? UserId { get; set; }
    public string? DeviceId { get; set; }
    public bool IsSynced { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

public enum TransactionType
{
    Income = 0,
    Expense = 1
}

File: lib/core/models/transaction.dart
import 'package:hive/hive.dart';

part 'transaction.g.dart';

@HiveType(typeId: 0)
class Transaction extends HiveObject {
  @HiveField(0)
  String id;

  @HiveField(1)
  String title;

  @HiveField(2)
  double amount;

  @HiveField(3)
  String category;

  @HiveField(4)
  DateTime date;

  @HiveField(5)
  TransactionType type;

  @HiveField(6)
  String? description;

  @HiveField(7)
  String? userId;

  @HiveField(8)
  String? deviceId;

  Transaction({
    required this.id,
    required this.title,
    required this.amount,
    required this.category,
    required this.date,
    required this.type,
    this.description,
    this.userId,
    this.deviceId,
  });

  Map<String, dynamic> toJson() => {
    'id': id,
    'title': title,
    'amount': amount,
    'category': category,
    'date': date.toIso8601String(),
    'type': type.index,
    'description': description,
    'userId': userId,
    'deviceId': deviceId,
  };

  factory Transaction.fromJson(Map<String, dynamic> json) => Transaction(
    id: json['id'],
    title: json['title'],
    amount: json['amount'].toDouble(),
    category: json['category'],
    date: DateTime.parse(json['date']),
    type: TransactionType.values[json['type']],
    description: json['description'],
    userId: json['userId'],
    deviceId: json['deviceId'],
  );
}

@HiveType(typeId: 1)
enum TransactionType {
  @HiveField(0)
  income,
  @HiveField(1)
  expense,
}